### ISSUE TITLE
Port zim to Gtk3 and Python3

### ISSUE THREAD

**Issue by jaap-karssenberg** [jaap-karssenberg/zim-desktop-wiki#152]
For future proofing we need to move to Gtk3 and to Python3

See project 1 (https://github.com/jaap-karssenberg/zim-desktop-wiki/projects/1) with refactoring actions to clean up the code base

Intermediate steps
- [x] Move to the `gi` library instead of `pygtk` - go to Gtk3
  - [x] Ensure full test coverage of Gtk interfaces to find regressions !
- [x] Within python 2.7 create a python 3 ready code base -- keep this branch alive to allow python2 releases with fixes if needed
- [x] Move to python 3
  - [ ] Manual testing of all parts not covered by test suite !
- [ ] Remove deprecated Gtk widgets

Once this is completed we could support a more Gnome3 consistent interface (optional, keep classic interface for Windows and other desktops)

**Comment by genodeftest** [comment]
For porting to `gi`, there is a script named [`pygi-convert.sh`](https://git.gnome.org/browse/pygobject/tree/pygi-convert.sh) in the PyGObject repo.

**Comment by mehdisadeghi** [comment]
Let's do this, I can help!

**Comment by jaap-karssenberg** [comment]
Converting to `gi` now, may take till after the x-mas holiday to publish it (watch the `next` branch)

Once the big step is done, definitely could use help testing and cleaning up remaining issues

**Comment by jaap-karssenberg** [comment]
See also https://github.com/jaap-karssenberg/zim-desktop-wiki/wiki/Planning

**Comment by genodeftest** [comment]
@jaap-karssenberg wrote:
> Once the big step is done, definitely could use help testing and cleaning up remaining issues

Just leave a note here and I'm happy to help.

**Comment by muguu** [comment]
Just wondering, if you are planning to use glade?

**Comment by jaap-karssenberg** [comment]
Ongoing, see branch "next"

**Comment by jaap-karssenberg** [comment]
@muguu: no first want to port what we have, rewriting bits using Gtk.Builder can be done later, but is additional work not needed now

**Comment by jaap-karssenberg** [comment]
Closing this bug and opening separate bugs per porting issue

**Comment by mystyc** [comment]
Does the Zim-Next branch use the 32-bit or the 64-bit version of python3?

**Comment by jaap-karssenberg** [comment]
@mystyc as far as I'm aware any version of python3 should work, I provided
sources only, no compiled libraries

**Comment by genodeftest** [comment]
The `next` branch works nicely for me. There is only one issue I've seen so far: The ToC widget (zim.plugins.tableofcontents.FloatingToC) is too small, rendering it hard to read. Should I open a separate bug report for that?

**Comment by genodeftest** [comment]
Answering my previous comment: This is already reported as #328.
